slo-workload/src/Commands/RunCommand.php

```diff
@@ -0,0 +1,421 @@
+                use ($query, $dataGenerator, $tableName, &$attemps) {
+                    $attemps++;
+                    return $session->query($query, DataGenerator::getUpsertData());
+                }, true, null, [
+                    'callback_on_error' => function (\Exception $e) {
+                        Utils::retriedError($this->queueId, 'write', get_class($e));
+                    }
+                ]);
+                Utils::metricDone("write", $this->queueId, $attemps, (microtime(true) - $begin) * 1000);
+            } catch (\Exception $e) {
+                if ($attemps == 0) $attemps++;
+                $table->getLogger()->error($e->getMessage());
+                Utils::metricFail("write", $this->queueId, $attemps, get_class($e), (microtime(true) - $begin) * 1000);
+            } finally {
+                $delay = ($begin - microtime(true)) * 1e6 + 1e6 / Defaults::RPS_PER_READ_FORK;
+                usleep($delay > 0 ? $delay : 1);
+            }
+        }
+    }
+
+    protected function metricsJob(int $reportPeriod, int $time, float $startTime, string $url, int $queueId)
+    {
+        $registry = new CollectorRegistry(new InMemory);
+        $pushGateway = new \PrometheusPushGateway\PushGateway($url);
+
+        $latencies = $registry->getOrRegisterSummary('', 'latency', 'summary of latencies in ms', ['jobName', 'status'], 15, [0.5, 0.99, 0.999]);
+        $oks = $registry->getOrRegisterGauge('', 'oks', 'amount of OK requests', ['jobName']);
+        $notOks = $registry->getOrRegisterGauge('', 'not_oks', 'amount of not OK requests', ['jobName']);
+        $inflight = $registry->getOrRegisterGauge('', 'inflight', 'amount of requests in flight', ['jobName']);
+        $errors = $registry->getOrRegisterGauge('', 'errors', 'amount of errors', ['jobName', 'class', 'in']);
+        $attempts = $registry->getOrRegisterHistogram('', 'attempts', 'summary of amount for request', ['jobName', 'status'], range(1, 10, 1));
+        $msgQueue = msg_get_queue($queueId);
+
+        $registry->wipeStorage();
+        $pushGateway->push($registry, 'workload-php', [
+            'sdk' => 'php',
+            'sdkVersion' => Ydb::VERSION
+        ]);
+        $pushGateway->delete('workload-php', [
+            'sdk' => 'php',
+            'sdkVersion' => Ydb::VERSION
+        ]);
+
+        $lastPushTime = microtime(true);
+
+        foreach ($this->errors as $error) {
+            $errors->incBy(0, ['read', $error, 'finally']);
+            $errors->incBy(0, ['write', $error, 'finally']);
+            $errors->incBy(0, ['read', $error, 'retried']);
+            $errors->incBy(0, ['write', $error, 'retried']);
+        }
+
+        $pushGateway->push($registry, "workload-php", [
+            'sdk' => 'php',
+            'sdkVersion' => Ydb::VERSION
+        ]);
+
+
+        while (microtime(true) <= $startTime + $time) {
+            while (msg_receive($msgQueue, 1, $msgType, 1024, $message)) {
```

Comment:
Use 0 for receive all messages or named constant MSG_TYPE for message type, not magic '1'.

use PHP_INT_MAX for effective disable message size limit.

should save error code to some variable and log it.

No need retries if msg_receive return false - need to die with error.